server/src/main/java/org/elasticsearch/index/engine/RecoverySourcePruneMergePolicy.java

```diff
@@ -39,18 +39,22 @@
 import java.util.function.Supplier;
 
 final class RecoverySourcePruneMergePolicy extends OneMergeWrappingMergePolicy {
-    RecoverySourcePruneMergePolicy(String recoverySourceField, Supplier<Query> retainSourceQuerySupplier, MergePolicy in) {
+    RecoverySourcePruneMergePolicy(String recoverySourceField, String idField, Supplier<Query> retainSourceQuerySupplier, MergePolicy in) {
```

Comment:
Maybe `idFieldToPrune`. It's not just the name of the `_id` field - it's non-null if you want to prune it. Heck, it could be a `boolean` if you want. Maybe better if it's a boolean actually.

Reply:
Yes, we would never trim any other field and _id is always hardcoded to `_id`. So a boolean parameter will work fine here.